perfectextractor/apps/extractor/base.py

```diff
@@ -68,22 +91,30 @@ def __init__(self, language_from, languages_to=None,
                 config.read_file(config_file)
             except AttributeError:  # Support for old method (Python < 3.2)
                 config.readfp(config_file)
-            self.config = config
+            self.config = CachedConfig(config)
 
         # Other variables
         self.other_extractors = []
         self.alignment_xmls = dict()
 
-    def process_folder(self, dir_name):
+    def process_folder(self, dir_name, progress_cb=None, done_cb=None):
         """
         Creates a result file and processes each file in a folder.
         """
+        progress_total = len(self.collect_file_names(dir_name))
```

Comment:
`collect_file_names` is called in `generate_results` as well, potentially save this to a variable?